### PR TITLE
Feature/undo

### PR DESCRIPTION
**Pull request by jamarob**
- small layout improvements
- undo functionality for last user action
- delete notes without asking permission

### CODE REVIEW

File: src/components/UndoMessage.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  color: whitesmoke;
+  width: 100%;
+  top: 100px;
+  left: ${props => (props.show ? 0 : '-105%')};
+  box-shadow: 2px 2px 2px 2px lightgray;
+  transition: 0.3s left ease-in-out;
+
+  button {
+    font-size: 16px;
+    background: none;
```
Comment (carinamenke) [inline src/components/UndoMessage.js]: Do you need to mention `background: none` explicitly? Does it maybe also work  by default if you leave this line out?

File: src/components/DateView.js [hunk 2]
```diff
@@ -10,22 +12,26 @@ DateView.propTypes = {
...
-function toLocaleString(time) {
-  return new Date(time).toLocaleString()
-}
-
 const StyledDate = styled.div`
   color: gray;
   font-size: 14px;
+  display: flex;
+  align-items: center;
+  gap: 4px;
```
[the remark this answers is not in the document]
Comment (jamarob) [reply src/components/DateView.js]: I don't know either.. somebody else already commented on this. But it works, is documented on mdn and I saw Jerry use it, so guess it is okay :)

File: src/components/NoteForm.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ export default function NoteForm({ onSave, text }) {
   }
 }
 
-const StyledNoteForm = styled.section`
+const StyledNoteForm = styled.main`
```
[the remark this answers is not in the document]
Comment (jamarob) [reply src/components/NoteForm.js]: The browser does not prevent you from inserting multiple main elements, but it breaks the semantics, as there should be only one main element. In a SPA, where you switch between screens, this shouldn't be a problem as there is only one main element shown at a time.

File: src/components/Search.js [hunk 4]
```diff
@@ -3,53 +3,47 @@ import styled from 'styled-components/macro'
...
 }
 
-export default Search
-
 const StyledSearch = styled.section`
   display: flex;
-  margin: 8px 16px 0 16px;
+  padding: 8px;
   gap: 4px;
+  box-shadow: 2px 2px 2px 2px lightgray;
```
[the remark this answers is not in the document]
Comment (jamarob) [reply src/components/Search.js]: This is a must! :D And that's exactly what I will do in the next PR.

File: src/components/UndoMessage.js (hunk 1 above)
Comment (jamarob) [reply src/components/UndoMessage.js]: The default background is a greyish color, I wanted to have an outlined look, where the background of the parent element shines through.

**Pull request merged**